### PR TITLE
SPE-2006: Add treatment failure blame routing

### PR DESCRIPTION
**Pull request by JamesJedi420**
## Summary

- Adds pure deterministic `buildTreatmentFailureBlameRoutingReport` for treatment-failure accountability routing (SPE-2006).
- Linear: https://linear.app/spectranoir/issue/SPE-2006/batch-9977-treatment-failure-blame-routing-system
- Blocks unsupported subject-side deflection when material failure evidence exists; requires treatment-limitation acknowledgment for institutional routes.

## Review hardening (ee3ad0c)

- Week-specific attributions only pair exact-week or week-agnostic failure contexts (no cross-week fallback).
- Weekless attributions require acknowledgments that match the resolved context week (or are week-agnostic).
- Invalid contexts with blank `subjectId`/`protocolId` are excluded from `failureContextCount`.
- Material-failure signal membership uses one precomputed `Set` per report.
- Institutional-required findings dedupe by `contextId`; week on finding uses `attribution.week ?? context.week`.
- Blank subject/protocol attributions excluded before summary (no phantom unpaired counts).

## Smallest boundary

- New domain helper + fixture tests only.
- No GameState, weekly tick, UI, SPE-2003/SPE-2010 imports, doctrine, scorecard, or policy selector.

## Files changed

- `src/domain/treatmentFailureBlameRouting.ts`
- `src/test/treatmentFailureBlameRouting.test.ts`

## Validation

- `npm run test:run -- src/test/treatmentFailureBlameRouting.test.ts` (27 tests)
- `npm run test:run -- src/test/medicalOutcomeDeviationAudit.test.ts` (adjacent)
- `npm run test:run -- src/test/staffTreatmentTelemetry.test.ts` (adjacent)
- `npm run test:run` (3145 tests)
- `npm run lint`
- `git diff --check`

## Gap / edge-case / boundary audit

| Category | Result |
| --- | --- |
| Blame routing | Subject deflection blocked without acknowledgment; institutional routes require acknowledgment; staff route approved with material failure; missing_observation not material; unrelated-week context cannot drive routing |
| Pairing | subject/protocol/week deterministic; weekless attribution uses context week for acknowledgment; invalid contexts excluded; institutional dedupe by contextId |
| Determinism | Stable sort, lines, summary; inputs not mutated |
| Boundaries | No GameState, SPE-2003, SPE-2010, UI, batch strings |

## Out of scope

- GameState persistence, runtime integration, UI
- SPE-2003/SPE-2010 adapters (slice 2)
- SPE-2008 scorecard, SPE-2001 doctrine, SPE-2004 policy selector, SPE-2009 personnel actions
- Recovery/trauma projection, uncertainWorldState

## Test plan

- [x] Targeted SPE-2006 tests (27)
- [x] Adjacent deviation audit + staff telemetry tests
- [x] Full suite + lint

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: src/domain/treatmentFailureBlameRouting.ts [hunk 1]
```diff
@@ -0,0 +1,820 @@
...
+      attribution.protocolId,
+      acknowledgment.subjectId,
+      acknowledgment.protocolId
+    )
+  ) {
+    return false
+  }
+  if (attribution.week === undefined) {
+    return true
+  }
```
[the remark this answers is not in the document]
Comment (JamesJedi420) [reply src/domain/treatmentFailureBlameRouting.ts]: Fixed in 512ddc9: resolveAcknowledgment now uses the resolved failure context week when attribution.week is omitted; acknowledgments from other weeks no longer approve routes.

File: src/domain/treatmentFailureBlameRouting.ts [hunk 2]
```diff
@@ -0,0 +1,820 @@
...
+  return target === 'staff_execution_gap'
+}
+
+function hasMaterialFailure(
+  context: TreatmentFailureContext,
+  materialSignals: readonly TreatmentFailureSignalKind[]
+): boolean {
+  const materialSet = new Set(materialSignals)
+  return context.failureSignals.some((signal) => materialSet.has(signal))
+}
```
[the remark this answers is not in the document]
Comment (JamesJedi420) [reply src/domain/treatmentFailureBlameRouting.ts:314]: Fixed in 512ddc9: resolveOptions builds materialFailureSignalSet once and all material-failure checks reuse it.

File: src/domain/treatmentFailureBlameRouting.ts [hunk 3]
```diff
@@ -0,0 +1,818 @@
...
+    }
+
+    const { week } = sample
+
+    const alreadyRequired = findings.some(
+      (finding) =>
+        finding.kind === 'institutional_accountability_required' &&
+        finding.subjectId === sample.subjectId &&
+        finding.protocolId === sample.protocolId &&
+        finding.week === week
```
[the remark this answers is not in the document]
Comment (JamesJedi420) [reply src/domain/treatmentFailureBlameRouting.ts]: Fixed in 512ddc9: institutional-required findings dedupe by contextId, and finding week uses sample.week ?? context.week.

File: src/domain/treatmentFailureBlameRouting.ts [hunk 4]
```diff
@@ -0,0 +1,842 @@
...
+          row.subjectId.length > 0 &&
+          row.protocolId.length > 0
+      ),
+    (row) => row.contextId
+  )
+  const proposedAttributions = dedupeById(
+    input.proposedAttributions
+      .map(normalizeAttribution)
+      .filter((row) => row.attributionId.length > 0),
+    (row) => row.attributionId
```
[the remark this answers is not in the document]
Comment (JamesJedi420) [reply src/domain/treatmentFailureBlameRouting.ts:727]: Fixed in ee3ad0c: proposed attributions with blank subjectId/protocolId are filtered during normalization (same as failure contexts), so they no longer inflate unpairedAttributionCount.
